python/TestHarness/schedulers/Scheduler.py

```diff
@@ -0,0 +1,539 @@
+        # to a soft limit. If however a max_processes  is passed we'll treat it as a hard limit.
+        # The difference is whether or not we allow single jobs to exceed
+        # the number of slots.
+        if params['max_processes'] == None:
+            self.job_slots = 1
+            self.soft_limit = True
+        else:
+            self.job_slots = params['max_processes'] # hard limit
+            self.soft_limit = False
+
+        # Requested average load level to stay below
+        self.average_load = params['average_load']
+
+        # Initialize runner pool based on available slots
+        self.tester_pool = ThreadPool(processes=self.job_slots)
+
+        # Initialize status pool to only use 1 process (to prevent status messages from getting clobbered)
+        self.status_pool = ThreadPool(processes=1)
+
+        # Thread Locking
+        self.thread_lock = threading.RLock()
+
+        # Workers in use
+        self.workers_in_use = 0
+
+        # A map containing our tester name to dag objects
+        self.name_to_dag = {}
+
+        # A simple queue increment to keep track of jobs
+        self.job_queue_count = 0
+
+    ## Run the command asynchronously
+    def run(self, job_container):
+        return
+
+    ## process test results
+    def returnToTestHarness(self, tester, thread_lock):
+        return
+
+    # Return post run command from derived classes
+    def postCommand(self):
+        return
+
+    # Allow derived schedulers to initiate findAndRunTests again
+    def goAgain(self):
+        return
+
+    ## Returns a list of tests to run
+    def getTests(self):
+        return
+
+    # Notify derived schedulers of status change
+    def notifySchedulers(self, tester):
+        return
+
+    # Return boolean on output ready to be read
+    def checkOutputReady(self, tester):
+        return True
+
+    # Call back to TestHarness to closeFiles in the event a derived scheduler wants to exit abnormally
+    def cleanup(self, exit_code=0):
+        self.harness.closeFiles(exit_code)
+
+    # return bool if we want to ignore prereq requirements
+    def skipPrereqs(self):
+        if self.options.ignored_caveats:
+            caveat_list = [x.lower() for x in self.options.ignored_caveats.split()]
+            if 'all' in self.options.ignored_caveats or 'prereq' in self.options.ignored_caveats:
+                return True
+        return False
+
+    # return a set of finished non-passing or will be skipped tests
+    def skippedTests(self, testers):
+        skipped_failed = set([])
+        for tester in testers:
+            if (tester.isFinished() and not tester.didPass()) \
+               or not tester.getRunnable(self.options):
+                skipped_failed.add(tester)
+        return skipped_failed
+
+    # Delete and return downstream nodes based on tester status and TestHarness options
+    # TODO: change the method name, as it doesn't always delete downstream tests
```

Comment:
Well, I'd suggest that you go ahead and rename it then: Maybe `processDownstreamTests`?